### ISSUE TITLE
Store-gateway runs into panic

### ISSUE THREAD

**Issue by agermain** [grafana/mimir#2959]
#### Describe the bug
Store gateway panics with an exception raised by a minio library.

#### To Reproduce
Steps to reproduce the behavior: We are running `mimir:r200-ae57256`, we don't have a reproducer, however we are running a very basic configuration of store gateway:
```
    store_gateway:
      persistentVolume:
        enabled: true
        size: 100Gi
      replicas: 2
```
It doesn't have any limits, so we know it's not an OOM issue, just wondering if this could be the handling of transient S3 failures.

#### Expected behavior

Expected store-gateway not to panic.

#### Environment

- Infrastructure: Kubernetes
- Deployment tool: helm

#### Additional Context
Logs:
```
{"log":"level=info ts=2022-09-15T05:13:22.554165588Z caller=gateway.go:288 msg=\"synchronizing TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T05:13:22.554244611Z"}
{"log":"level=info ts=2022-09-15T05:13:22.693616008Z caller=gateway.go:294 msg=\"successfully synchronized TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T05:13:22.693711652Z"}
{"log":"level=info ts=2022-09-15T05:29:43.442844584Z caller=gateway.go:288 msg=\"synchronizing TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T05:29:43.442969503Z"}
{"log":"level=info ts=2022-09-15T05:29:43.598777599Z caller=gateway.go:294 msg=\"successfully synchronized TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T05:29:43.598851503Z"}
{"log":"level=info ts=2022-09-15T05:46:04.331928567Z caller=gateway.go:288 msg=\"synchronizing TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T05:46:04.33201909Z"}
{"log":"level=info ts=2022-09-15T05:46:04.478577621Z caller=gateway.go:294 msg=\"successfully synchronized TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T05:46:04.478648781Z"}
{"log":"level=info ts=2022-09-15T06:02:25.221735261Z caller=gateway.go:288 msg=\"synchronizing TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T06:02:25.221850584Z"}
{"log":"level=info ts=2022-09-15T06:02:25.318177301Z caller=gateway.go:294 msg=\"successfully synchronized TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T06:02:25.318243708Z"}
{"log":"level=info ts=2022-09-15T06:18:46.110479058Z caller=gateway.go:288 msg=\"synchronizing TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T06:18:46.1105663Z"}
{"log":"level=info ts=2022-09-15T06:18:46.235677599Z caller=gateway.go:294 msg=\"successfully synchronized TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T06:18:46.235757355Z"}
{"log":"level=warn ts=2022-09-15T06:30:24.921059404Z caller=grpc_logging.go:64 duration=22.452905ms method=/gatewaypb.StoreGateway/Series err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000014\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:30:24.921426911Z"}
{"log":"level=warn ts=2022-09-15T06:30:24.974360919Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=82.865652ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000008\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:30:24.974429272Z"}
{"log":"level=warn ts=2022-09-15T06:30:24.977362572Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=59.396756ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000014\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:30:24.977423326Z"}
{"log":"level=warn ts=2022-09-15T06:30:24.977539992Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=57.176281ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000014\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:30:24.977588808Z"}
{"log":"level=warn ts=2022-09-15T06:30:24.979659528Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=18.128875ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000009\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:30:24.979719312Z"}
{"log":"level=info ts=2022-09-15T06:35:07.000199411Z caller=gateway.go:288 msg=\"synchronizing TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T06:35:07.000292886Z"}
{"log":"level=info ts=2022-09-15T06:35:07.134954737Z caller=gateway.go:294 msg=\"successfully synchronized TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T06:35:07.135028338Z"}
{"log":"level=warn ts=2022-09-15T06:35:37.716539107Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=6.039169ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000014\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:35:37.716606857Z"}
{"log":"level=warn ts=2022-09-15T06:35:37.751063887Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=69.354868ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000014\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:35:37.751168297Z"}
{"log":"level=warn ts=2022-09-15T06:35:37.754684822Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=40.746714ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000009\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:35:37.754749695Z"}
{"log":"level=warn ts=2022-09-15T06:35:37.756487606Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=16.471086ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000009\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:35:37.756555633Z"}
{"log":"level=warn ts=2022-09-15T06:46:25.265555004Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=24.918214ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000019\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:46:25.265670892Z"}
{"log":"level=warn ts=2022-09-15T06:46:25.265720563Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=25.144417ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000019\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:46:25.265795738Z"}
{"log":"level=warn ts=2022-09-15T06:46:25.274405678Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=145.476073ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: read range for seq 18 offset 11624f48: context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:46:25.274506958Z"}
{"log":"level=warn ts=2022-09-15T06:46:25.281452737Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=88.019972ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: fast forward range reader: context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:46:25.281639299Z"}
{"log":"level=warn ts=2022-09-15T06:46:25.283023809Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=32.165493ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000019\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:46:25.283099798Z"}
{"log":"level=warn ts=2022-09-15T06:46:25.290164332Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=58.902795ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: fast forward range reader: context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:46:25.290221521Z"}
{"log":"level=warn ts=2022-09-15T06:46:31.103001153Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=193.804107ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: fast forward range reader: context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:46:31.103231298Z"}
{"log":"level=info ts=2022-09-15T06:51:27.889352925Z caller=gateway.go:288 msg=\"synchronizing TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T06:51:27.889435344Z"}
{"log":"level=info ts=2022-09-15T06:51:28.00386305Z caller=gateway.go:294 msg=\"successfully synchronized TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T06:51:28.003944211Z"}
{"log":"level=warn ts=2022-09-15T06:56:32.146214839Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=28.683899ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: get range reader: Get \\\"https://some-aws-bucket.s3.dualstack.us-east-2.amazonaws.com/tenant/01GCZD7C80TZS7SDDH7SE258PJ/chunks/000019\\\": context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T06:56:32.146331615Z"}
{"log":"level=info ts=2022-09-15T07:07:48.778333629Z caller=gateway.go:288 msg=\"synchronizing TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T07:07:48.778441001Z"}
{"log":"level=info ts=2022-09-15T07:07:48.907044365Z caller=gateway.go:294 msg=\"successfully synchronized TSDB blocks for all users\" reason=periodic\n","stream":"stderr","time":"2022-09-15T07:07:48.907117101Z"}
{"log":"panic: send on closed channel\n","stream":"stderr","time":"2022-09-15T07:11:44.924777583Z"}
{"log":"level=warn ts=2022-09-15T07:11:44.919597851Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=128.666013ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: fast forward range reader: context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T07:11:44.919729084Z"}
{"log":"level=warn ts=2022-09-15T07:11:44.919872437Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=110.17479ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: fast forward range reader: context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T07:11:44.919946906Z"}
{"log":"level=warn ts=2022-09-15T07:11:44.919903664Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=130.29575ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: fast forward range reader: context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T07:11:44.919973574Z"}
{"log":"level=warn ts=2022-09-15T07:11:44.91991113Z caller=grpc_logging.go:64 method=/gatewaypb.StoreGateway/Series duration=160.037249ms err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: read range for seq 18 offset 10588901: context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T07:11:44.919976414Z"}
{"log":"level=warn ts=2022-09-15T07:11:44.92024891Z caller=grpc_logging.go:64 duration=110.550622ms method=/gatewaypb.StoreGateway/Series err=\"rpc error: code = Aborted desc = fetch series for block 01GCZD7C80TZS7SDDH7SE258PJ: load chunks: fast forward range reader: context canceled\" msg=gRPC\n","stream":"stderr","time":"2022-09-15T07:11:44.920372069Z"}
{"log":"\n","stream":"stderr","time":"2022-09-15T07:11:44.924783751Z"}
{"log":"goroutine 1604297 [running]:\n","stream":"stderr","time":"2022-09-15T07:11:44.924785934Z"}
{"log":"github.com/minio/minio-go/v7.(*Object).doGetRequest(_, {{0xc007f76000, 0x3e80, 0x3e80}, 0x0, 0x0, 0x1, 0x0, 0x1, 0x0, ...})\n","stream":"stderr","time":"2022-09-15T07:11:44.924791304Z"}
{"log":"\u0009/__w/mimir/mimir/vendor/github.com/minio/minio-go/v7/api-get-object.go:319 +0xdc\n","stream":"stderr","time":"2022-09-15T07:11:44.924793735Z"}
{"log":"github.com/minio/minio-go/v7.(*Object).Read(0xc000a42700, {0xc007f76000?, 0x77?, 0xc00676bb78?})\n","stream":"stderr","time":"2022-09-15T07:11:44.924798014Z"}
{"log":"\u0009/__w/mimir/mimir/vendor/github.com/minio/minio-go/v7/api-get-object.go:399 +0x248\n","stream":"stderr","time":"2022-09-15T07:11:44.924800286Z"}
{"log":"github.com/thanos-io/thanos/pkg/objstore.(*timingReadCloser).Read(0xc009a8fea0, {0xc007f76000?, 0x77?, 0xc00676bbf0?})\n","stream":"stderr","time":"2022-09-15T07:11:44.924814692Z"}
{"log":"\u0009/__w/mimir/mimir/vendor/github.com/thanos-io/thanos/pkg/objstore/objstore.go:651 +0x32\n","stream":"stderr","time":"2022-09-15T07:11:44.924819386Z"}
{"log":"github.com/thanos-io/thanos/pkg/objstore.(*tracingReadCloser).Read(0xc0051e3500, {0xc007f76000?, 0x0?, 0x7f474082d108?})\n","stream":"stderr","time":"2022-09-15T07:11:44.924821199Z"}
{"log":"\u0009/__w/mimir/mimir/vendor/github.com/thanos-io/thanos/pkg/objstore/tracing.go:138 +0x32\n","stream":"stderr","time":"2022-09-15T07:11:44.924826727Z"}
{"log":"bufio.(*Reader).Read(0xc0059ea5a0, {0xc007f7a000, 0x54, 0x44ff72?})\n","stream":"stderr","time":"2022-09-15T07:11:44.924829212Z"}
{"log":"\u0009/usr/local/go/src/bufio/bufio.go:236 +0x1b4\n","stream":"stderr","time":"2022-09-15T07:11:44.92483248Z"}
{"log":"io.ReadAtLeast({0x2564000, 0xc0059ea5a0}, {0xc007f7a000, 0x54, 0x3e80}, 0x54)\n","stream":"stderr","time":"2022-09-15T07:11:44.924855323Z"}
{"log":"\u0009/usr/local/go/src/io/io.go:331 +0x9a\n","stream":"stderr","time":"2022-09-15T07:11:44.924858215Z"}
{"log":"io.ReadFull(...)\n","stream":"stderr","time":"2022-09-15T07:11:44.924862086Z"}
{"log":"\u0009/usr/local/go/src/io/io.go:350\n","stream":"stderr","time":"2022-09-15T07:11:44.924866054Z"}
{"log":"github.com/grafana/mimir/pkg/storegateway.(*bucketChunkReader).loadChunks(0xc005f73920, {0x257f680, 0xc009af4200}, {0xc008114000, 0x109, 0xc000cb3870?}, {0x0, 0x0, 0x0}, 0x12, ...)\n","stream":"stderr","time":"2022-09-15T07:11:44.924893254Z"}
{"log":"\u0009/__w/mimir/mimir/pkg/storegateway/bucket.go:2656 +0x5dd\n","stream":"stderr","time":"2022-09-15T07:11:44.924897644Z"}
{"log":"github.com/grafana/mimir/pkg/storegateway.(*bucketChunkReader).load.func3()\n","stream":"stderr","time":"2022-09-15T07:11:44.924899993Z"}
{"log":"\u0009/__w/mimir/mimir/pkg/storegateway/bucket.go:2592 +0xa5\n","stream":"stderr","time":"2022-09-15T07:11:44.924906353Z"}
{"log":"golang.org/x/sync/errgroup.(*Group).Go.func1()\n","stream":"stderr","time":"2022-09-15T07:11:44.924908763Z"}
{"log":"\u0009/__w/mimir/mimir/vendor/golang.org/x/sync/errgroup/errgroup.go:75 +0x64\n","stream":"stderr","time":"2022-09-15T07:11:44.924910848Z"}
{"log":"created by golang.org/x/sync/errgroup.(*Group).Go\n","stream":"stderr","time":"2022-09-15T07:11:44.924912936Z"}
{"log":"\u0009/__w/mimir/mimir/vendor/golang.org/x/sync/errgroup/errgroup.go:72 +0xa5\n","stream":"stderr","time":"2022-09-15T07:11:44.924921274Z"}
```

**Comment by pracucci** [comment]
Looks like has been fixed by https://github.com/minio/minio-go/pull/1670.

**Comment by pracucci** [comment]
Mimir `main` already vendors Minio 7.0.30, so this issue should be fixed by upgrading to the upcoming Mimir 2.3.0. I'm going to close this issue, but feel free to reopen if Mimir 2.3.0 doesn't fix it.

**Comment by agermain** [comment]
`r200-ae57256` is currently showing:
```
Mimir, version 2.3.0 (branch: r200, revision: ae57256)
```
Are you referring to a specific revision/tag?

**Comment by pracucci** [comment]
I've rewritten the stack trace in a more readable format:

```
panic: send on closed channel

goroutine 1604297 [running]:
github.com/minio/minio-go/v7.(*Object).doGetRequest(_, {{0xc007f76000, 0x3e80, 0x3e80}, 0x0, 0x0, 0x1, 0x0, 0x1, 0x0, ...})
    /__w/mimir/mimir/vendor/github.com/minio/minio-go/v7/api-get-object.go:319 +0xdc
github.com/minio/minio-go/v7.(*Object).Read(0xc000a42700, {0xc007f76000?, 0x77?, 0xc00676bb78?})
    /__w/mimir/mimir/vendor/github.com/minio/minio-go/v7/api-get-object.go:399 +0x248
github.com/thanos-io/thanos/pkg/objstore.(*timingReadCloser).Read(0xc009a8fea0, {0xc007f76000?, 0x77?, 0xc00676bbf0?})
    /__w/mimir/mimir/vendor/github.com/thanos-io/thanos/pkg/objstore/objstore.go:651 +0x32
github.com/thanos-io/thanos/pkg/objstore.(*tracingReadCloser).Read(0xc0051e3500, {0xc007f76000?, 0x0?, 0x7f474082d108?})
    /__w/mimir/mimir/vendor/github.com/thanos-io/thanos/pkg/objstore/tracing.go:138 +0x32
bufio.(*Reader).Read(0xc0059ea5a0, {0xc007f7a000, 0x54, 0x44ff72?})
    /usr/local/go/src/bufio/bufio.go:236 +0x1b4
io.ReadAtLeast({0x2564000, 0xc0059ea5a0}, {0xc007f7a000, 0x54, 0x3e80}, 0x54)
    /usr/local/go/src/io/io.go:331 +0x9a
io.ReadFull(...)
    /usr/local/go/src/io/io.go:350
github.com/grafana/mimir/pkg/storegateway.(*bucketChunkReader).loadChunks(0xc005f73920, {0x257f680, 0xc009af4200}, {0xc008114000, 0x109, 0xc000cb3870?}, {0x0, 0x0, 0x0}, 0x12, ...)
    /__w/mimir/mimir/pkg/storegateway/bucket.go:2656 +0x5dd
github.com/grafana/mimir/pkg/storegateway.(*bucketChunkReader).load.func3()
    /__w/mimir/mimir/pkg/storegateway/bucket.go:2592 +0xa5
golang.org/x/sync/errgroup.(*Group).Go.func1()
    /__w/mimir/mimir/vendor/golang.org/x/sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
    /__w/mimir/mimir/vendor/golang.org/x/sync/errgroup/errgroup.go:72 +0xa5
```

**Comment by pracucci** [comment]
I opened an issue in Minio:
https://github.com/minio/minio-go/issues/1695

@agermain would you have time to take a look? Maybe it's not too difficult to fix.

**Comment by agermain** [comment]
> I opened an issue in Minio: [minio/minio-go#1695](https://github.com/minio/minio-go/issues/1695)
> 
> @agermain would you have time to take a look? Maybe it's not too difficult to fix.

Seems like it was fixed as part of https://github.com/minio/minio-go/pull/1679 released in v7.0.34. Will wait for the upstream upgrade of minio and report back, thanks for looking into this!

**Comment by agermain** [comment]
FYI, this issue has been addressed, been running https://github.com/grafana/mimir/commit/f2ad13345ef17f7892740c2623d998efe5d769d7 with no issues! Thanks @pracucci.
